### ISSUE TITLE
All of my requests in one page. Developer, please read this!

### ISSUE THREAD

**Issue by mvfsullivan** [farmerbb/Taskbar#38]
Firstly, I'd like to say thank you SO much for putting so much effort into such a fantastic app. TaskBar has turned my Android device into a desktop version of Android, and it is absolutely stunning.  

I truly believe in this app. It is unlike any other multi window / taskbar app by a hundred miles, so here are some things that I've come up with:

1. Have the taskbar apps consistently placed, including folder support. It's confusing when the apps change places based on last used.  Setting the "Display apps launched" to "After Taskbar start" and "Sort order" to "Most time used (descending) helps remove the clutter, but they still shift every now and then. I'd like to pin certain apps to the task bar and have them stay in place. If I could combine pinned apps to the task bar, as well as apps beside them ordered by most time used, it would be perfect (and exactly like Windows if you think about it, but even better).

2. Add buttons in the top bar to resize the window (this icon could be on the very left side), align the window (top, bottom, and freeform), minimize, maximize, and close (the last two are there already which is fantastic).

- As of right now, resizing windows is pretty much a guessing game. I love that you allowed us to resize the vertical size from the top AND the bottom, and I'm thankful for the ability to resize the windows in general, but it could use a dedicated resize mode with its own button in the top bar.
- If app is minimized, the icon on the taskbar is highlighted or has an indication that it is running and minimized (similar to what Windows does)
- If an app is maximized, instead of being truly full screen like stock Android does now, it would be perfect if the app actually ALWAYS had the top bar showing, and essentially just filled the screen. With the top bar always showing, we would be able to instantly align, resize, minimize, maximize, and close the app.
*** So true full window mode should essentially not be a thing if the user decides to enable the "always show the menu bar" option.

3. Please, please add resizable widgets! 

Not manual pre-set sizes for the entire homescreen.  
Although I'm incredibly thankful to even be able to set Taskbar as the launcher and add widgets, they could also use some work. I would suggest checking out the Nova Launcher and seeing how they handle widget sizing.

So to recap: Make the taskbar more static so apps dont move around, possibly making a hybrid static + most used combination. Add resize, align, and minimize buttons in the top bar, again, similar to Desktop Windows OS. 

This app has so much potential, it's incredible. I will donate a big chunk of money if these features are added!! That is an absolute promise my friend. I've been looking for something like this for YEARS.

**Comment by mvfsullivan** [comment]
I can see that your app icon is very well made so I assume you have excellent experience with Photoshop or graphic design, but if you ever want a second opinion and some free help, I would love to contribute, completely free, no strings. I love helping developers that create excellent apps, and you are absolutely one of them!

**Comment by farmerbb** [comment]
Thanks for your suggestions!

1. Understandable that you are wanting the app icons to remain consistently in place.  The main reason why the behavior is the way it is, is because there's no way for Taskbar to determine whether or not an app is actively running while it is a non-system app, so keeping the app icons in the same position ultimately wouldn't be useful for the end user.

That having been said, Taskbar does have this ability while it is installed as a system app, so I should definitely look into keeping the icons in place for this particular scenario.

2. Unfortunately, Taskbar does not have any control over the top bar (this is drawn by the Android system), and Taskbar also does not have any control over how apps are positioned outside of the initial launch (so minimizing windows and maximizing/resizing them after they are launched, sadly, is a no-go)

3. A lot of people have asked for true resizable widgets, it's just a matter of dedicating the time and resources to implementing this.  But I'll consider it noted!

Also, about the app icon, this was actually a third-party contribution (I'm no graphic designer).  With Android O's support for adaptive icons, though, it might be time to give it a refresh :)

**Comment by mvfsullivan** [comment]
I was suspicious that TaskBar utilized stock functions, so my requests would more or less be difficult to implement.

Have you considered having a root-enabled alternative that can take advantage of the power in order to incorporate such features?

All I really request are having some sort of static taskbar list (even if that means no hybrid static / active), and then being able to maximize (fit to screen, not 100% stock full screen) or minimize with some sort of taskbar indication. The static / hybrid taskbar, widgets, and advanced resizing toolbar are just luxury.

Those are the two I really really need. 

This app is incredible, and COMPLETELY changes the way I can use my smartphone.

**Comment by mvfsullivan** [comment]
Looks like you added full screen (maximized size, not true full screen), excellent work!
